cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/NoOpMetadataStore.java

```diff
@@ -117,13 +118,10 @@ public void removeTags(MetadataScope scope, MetadataEntity metadataEntity, Set<S
   }
 
   @Override
-  public MetadataSearchResponseV2 search(String namespaceId, String searchQuery,
-                                         Set<EntityTypeSimpleName> types,
-                                         SortInfo sort, int offset, int limit, int numCursors, String cursor,
-                                         boolean showHidden, Set<EntityScope> entityScope) {
-    return new MetadataSearchResponseV2(sort.toString(), offset, limit, numCursors, 0,
-                                        Collections.emptySet(),
-                                        Collections.emptyList(), showHidden, entityScope);
+  public MetadataSearchResponseV2 search(SearchRequest request) {
+    return new MetadataSearchResponseV2(request.getSortInfo().toString(), request.getOffset(), request.getLimit(),
```

Comment:
should we add a constructor in MetadataSearchResponseV2 which takes the SearchRequest and other stuff which is not in search request to avoid all this getters.

Reply:
this would require moving SearchRequest into proto, which might make sense if we had a client somewhere for search, but may otherwise be confusing since everything else there usually represents part of a request or response body.